### ISSUE TITLE
feat: The requirement is to add user management-related functions

### ISSUE THREAD

**Issue by china-powers** [apitable/apitable#637]
1. "User maintenance functions include adding, deleting, modifying, and querying user information."
2. "Is it possible to disable user registration?"
3. "Is it possible to remove the email verification code requirement for changing passwords?"

## Why? 

1. "Managing registered users on the website is inconvenient. It is not possible to view registered users on the website, and administrators cannot reset user passwords from the backend."
2. "It is necessary to be able to disable user registration, and only allow administrators to add new users from the backend. Alternatively, users can register from the frontend, but their accounts need to be approved by administrators before they can use them."
3. "The system operates in a completely offline environment, and it is not possible to send verification codes via email. Since our machines operate on an internal network, the verification code functionality needs to be removed."

**Comment by china-powers** [comment]
#582 #617

**Comment by Alex-hufeng** [comment]
"User maintenance functions include adding, deleting, modifying, and querying user information."
——————————
![CleanShot 2023-06-06 at 17 21 01@2x](https://github.com/apitable/apitable/assets/54580109/f625eb5e-d12d-4de8-823a-4de1f091ab23)
Do you mean this page? 



"Is it possible to disable user registration?"
"Is it possible to remove the email verification code requirement for changing passwords?"
——————————

We can provide partial customization function in Enterprise Edtion, you can submit the form below:
https://apitable.com/share/shrdaGGppsfg3pjQLXALG/fomnfpJ9XBfDDz6Dkz?fldy5ZmHYGZx2=self-hosted-pxpage

**Comment by china-powers** [comment]
> "User maintenance functions include adding, deleting, modifying, and querying user information." —————————— ![CleanShot 2023-06-06 at 17 21 01@2x](https://user-images.githubusercontent.com/54580109/243646410-f625eb5e-d12d-4de8-823a-4de1f091ab23.jpg) Do you mean this page?
> 
> "Is it possible to disable user registration?" "Is it possible to remove the email verification code requirement for changing passwords?" ——————————
> 
> We can provide partial customization function in Enterprise Edtion, you can submit the form below: https://apitable.com/share/shrdaGGppsfg3pjQLXALG/fomnfpJ9XBfDDz6Dkz?fldy5ZmHYGZx2=self-hosted-pxpage

Yes, SMTP cannot be used in private network environments. It is also not possible to disable the registration function. And administrators cannot manage other registered users.

**Comment by AubinF** [comment]
I fully agree with @MX8CLUB. I am serving a self-hosted instance of apitable, and I am a little bit surprised that it is not possible to disable the registration function. It is a must! Without it, anyone could create an account on any self-hosted instance.

**Comment by zacharya19** [comment]
**Walkaround**
Sadlly, it's not supported.
I used my K8S ingress to deny access to `/api/v1/register` path.
Note that it will prevent you from allowing new users to join.

My setup with ALB:
Annoation
```
alb.ingress.kubernetes.io/actions.response-404: >
      {"type":"fixed-response","fixedResponseConfig":{"contentType":"text/plain","statusCode":"404","messageBody":"Nothing here"}}
```

Path
```
          - pathType: Prefix
            path: /api/v1/register
            backend:
              service:
                name: response-404
                port:
                  name: use-annotation
```

**Comment by mr-kelly** [comment]
We completely understand your concerns regarding security and user administration in self-hosted environments. However, advanced identity access management features—such as disabling open registration, administrative password resets without email, LDAP/SSO integrations, and a dedicated admin console—are maintained as core capabilities of our Commercial/Enterprise editions to meet strict corporate compliance requirements. The open-source version is designed for basic collaborative use out-of-the-box. We recommend utilizing network-level ingress rules (like Nginx/K8s rules) to block the `/api/v1/register` endpoint as a workaround for the Community Edition. We will close this issue for now.

---

我们完全理解您对私有化部署环境中的安全性和用户管理的担忧。然而，高级的身份访问管理功能——如禁用开放注册、无需邮件的管理员密码重置、LDAP/SSO 集成以及专门的管理后台——是作为我们商业/企业版的核心功能进行维护的，以满足严格的企业合规要求。开源版旨在提供开箱即用的基础协作功能。对于社区版，我们建议您使用网络层面的入口规则（如 Nginx/K8s 规则）来拦截 `/api/v1/register` 接口作为规避方案。我们将暂时关闭此 Issue。
